dask/base.py

```diff
@@ -70,7 +71,7 @@ def visualize(self, filename="mydask", format=None, optimize_graph=False, **kwar
 
         Examples
         --------
-        >>> x.visualize(filename='dask.pdf')  # doctest: +SKIP
+        >>> x.visualize(filename='dask')  # doctest: +SKIP
```

Comment:
Hrm, even though we choose to add an extension, I don't think that we should encourage users to do this.  I would rather that they be explicit.  This seems like good practice generally, and also that way if we choose to reverse this decision in the future (which seems possible) then fewer people will be upset with us.  

```suggestion
```

Reply:
Can you explain a bit more what you mean by "explicit"? Are you saying we should encourage users to use `.visualize(filename='dask', format='pdf')` over `.visualize(filename='dask.pdf')`, or is it the other way around?